src/deploy/functions/build.ts

```diff
@@ -311,6 +312,67 @@ function envWithTypes(rawEnvs: Record<string, string>): Record<string, string |
   return out;
 }
 
+// Utility class to make it more fluent to use proto.convertIfPresent
+// The class usese const lambdas so it doesn't loose the this context when
+// passing Resolver.resolveFoo as a proto.convertIfPresent arg.
+// The class also recognizes that if the input is not null the output cannot be
+// null.
+class Resolver {
```

Comment:
Is this a copy + paste or a move of what I wrote? Do we need the definition twice?